### DISCUSSION TITLE
Cruiser Manual?

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#480] [Q&A] [answered]
Hi, is there a manual for cruiser existing?
In general I can manage the app as it's mostly self explaining. However, sometimes I'm struggling a little bit which setup is better or what is the difference or which switch I have to use to receive this or that output.

i.e.:
- which one (algorithm) is best for (offline) routing.
- what do I have to do when my height doesn't fit (it look like I'm almost higher then reality - talking about 50m)
- What influence (good/bad/useful) has Kalman filter?
- Which setting is best for most real time calculations...
- An visual overview diagram which setting can be found where? 

Cheers, Dirk

**Comment by devemux86** [comment, marked as answer]
Hi @DIrgi,

Currently there is no manual as it takes a lot of work and Cruiser is constantly getting new features!

- BRouter (offline) is the best selection (enabled by default) for offline routing with too many options.

- Some devices may correct the altitude to be above mean sea level.
You can enable it in "Settings | Location | Advanced | Altitude | Mean sea level".

- Kalman filter can usually correct GPS errors and jumps.
Also, it can provide a smoother movement on the map.

- What do you mean, rerouting in navigation?

- Do you mean the elevation diagram in navigation and route planning?
In navigation it is in "Settings | Navigation | Display" or long press the direction icon.
In route planning it can be turned on / off in menu "Routing".

If you have more questions, please write them and we will help you! :slightly_smiling_face:

**Reply by DIrgi** [reply]
Hi Emux, thank you for your feedback. 


- Altitude:
Coming from a Garmin Navi (I want to get rid of it), now trying to do it on an outdoor tablet with your software. My tablet has Andorid 12 and unfortunately no further settings when it comes to altitude settings. However (my tablet has no alti- or barometer) when I use OSMand the displayed altitude is close to my location which is round about 100m above se level. OSMand shows me 101-103m . Orux Maps shows me a little bit less, about 93m - However Cruiser tells me 150m. Afaik if there is no altimeter hardware available, altitude is calculated from GPS data or map data (with DEM model?). I know that calculated figures have a certain deviation
I don't know how your algorithm looks like but might it be possible to have a closer calculation and if not to enter at least an offset parameter? 

- rerouting was not on me.

- diagram - sorry I meant something different (connected to the manual) more like a flow chart of the available functions. But since it's connected to the manual, you gave already the answer.

Cheers, Dirl

**Reply by devemux86** [reply]
@DIrgi 

Have you selected "**Mean sea level**" in "Settings | Location | Advanced | Altitude"?

It should return different altitude from the other (default) option.

**Reply by DIrgi** [reply]
Yes I already tried it before, but just went outside again and switched from "normal" to "mean". According to your instructions. 
At the moment I'm not in navigation mode so I have only as stand alone the altitude displayed on top right of the screen in front of the map. 
The result wit the 150m was the same BUT.....I noticed a different display for a split second it shows something with 103m which ist "real" enough - However I recognized like a occasional flickering with the altitude display - It seems at the same place there a two altitudes displayed at the same time but one overlays the other and that one with the 103m is in the background and the one with 150m in the foreground. Ok, we are on the right way but how it can be having two data set at the same place displayed? And the deal with the same output: The height...strange.

**Reply by devemux86** [reply]
@DIrgi thanks for the tests.

What have you selected in "Settings | Location | Location service"?
Android or Google? Can you try the other option?

**Reply by DIrgi** [reply]
OK, this is what I tried right now.

presets:
I'm standing on the balcony (having GPS available), connected to WLAN (when I navigate, I normally use offline navigation)

setting variants:
1.  when I use "normal" altitude the "wrong" altitude becomes displayed which is round about +50m to my real altitude
2.  when I use "mean level" then - yes - it seems more correct but as already mentioned I have a flipping(rolling) alt display, showing both at the same place but jumping from one to the other. However one is more dominant displayed and 99% of the time. The flipping interval is every one or two seconds but one altitude parameter is show only maybe a tenth of a second or even shorter.
I try to do some shots, not easy to hit the right time while switching. but I did.
See picture the altitudes.
3. What else do I recognize?  When the more stable displayed altitude is visible, then the navigation cursor is the know triangle. When there's the flip, it switches to a blue dot and the triangle disappear.
4. It is absolutely independent if I use either Google or Android as location service. Result is the same with both.

<img src="https://github.com/devemux86/cruiser/assets/57968526/db155edb-87e2-4cd6-a266-e3c907908cd5" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/57968526/93ebf422-14e3-4497-987c-4247e2a19e57" width="50%"/>

**Reply by devemux86** [reply]
@DIrgi 

2. Thanks for the report, I will check it. The altitude is provided by the sensor. I wonder if it sends (?) both values.
Do you have another device to make similar tests?

3. The location arrow appears when the GPS sends the location heading and it usually needs to have a speed.
It can be controlled in "Settings | Location | Advanced | Orientation threshold".

**Reply by devemux86** [reply]
@DIrgi 

> I have a flipping(rolling) alt display, showing both at the same place but jumping from one to the other. However one is more dominant displayed and 99% of the time. The flipping interval is every one or two seconds but one altitude parameter is show only maybe a tenth of a second or even shorter.

Is this with Kalman filter enabled? What happens without the Kalman filter?

Which value appears most often, the "correct" (mean sea level) or the other?

**Reply by devemux86** [reply]
@DIrgi

If you want, can you test the altitude in the latest Beta (with and without Kalman filter)?

- https://github.com/devemux86/cruiser/discussions/482

**Reply by DIrgi** [reply]
Ok, here we go...I did some more tests right now.
**Please don't get me wrong - I really don't wanna kidding you but now it seems that everything works fine or saying it in another way I cannot reproduce the issue.**

(using Cruiser 3.0.42 on an Android 12)

What did I do?
- Well I powered the tablet yesterday evening completely off. (But I do this all the time and I think I did this as well yesterday when I recognized the issue)
- right now I restarted the tablet, tablet connected to WLAN. I went out to balcony, started Cruiser again, received round about 25 satellites (I can receive Glonass, BaiDou, GPS and Galileo)
- I switches multiple time between "normal" and "mean" altitude setting where "mean" is the most accurate.
- I switched multiple times between "Android" and "Google" service where I recognized a slight difference of up to 3m - So finally to neglect. btw. what is the difference between both? Does one need an online connection?
- I switched as well Kalman Filter on/off several times (all in connection with the upper settings - So to test every possible situation) - Finally Kalman had no influence
- And I switched between a routing on and off mode (but I didn't move)

At the end I could not reproduce the issue from yesterday which is finally good and everything works well on one hand but bad because we don't know the reason for. Yesterday and now, Cruiser was the only app which was running (except the needed android background system apps) So I really feel sorry to panic you - I really have no glue what caused the problem but trust me, it was a described.

Best regards, Dirk

**Reply by devemux86** [reply]
@DIrgi :+1: 

Thanks very much for the extensive tests and description!

I asked you if want to test the latest Beta, because it contains a minor altitude improvement.
But it's nice if it's resolved by itself. :slightly_smiling_face: 
Usually a device restart helps to reinitialize the sensors.

Many thanks again and please report if you see anything else.

---

The difference between the Android vs Google location services:
- **Android** location service: the default native location service we've known since the beginning of Android.

- **Google** location service: it is advertised by Google as "battery-efficient" and "combines different signals to provide the location information". But this could also mean a not native **GPS** location.

So it is best to try both location services and use the one that works better with your device.

**Comment by devemux86** [comment]
@DIrgi 

There is a general rule in navigation and other screen panels and buttons:
- **Tap** the panels to activate more actions

- **Long press** the panels to open their options

**Reply by DIrgi** [reply]
Thanks, I'm aware about.

**Comment by DIrgi** [comment]
Some more proposals, suggestions wishes for an update:

Since I use a tablet, I have plenty of space for (visible) display compared to cell phone users. I'm a little bit of a data junkie and love to have several parameters available while driving. Most of them which are available  I use during navigation. However, there are some functions which allow a either/or selection only. i.e. time or distance to next waypoint (I know I can tap on to switch) or time/distance to final destination.  It would be great to have not just a either/or possibility but a "and", too - So display both at the same time on demand.

Will it be possible to have a all time ODOmeter? - Means a ODOmeter which is running the whole time as long as I stop or reset it - So finally even over a couple of days or even more - as long as the app is active and navigating and I finally stop it?

You mentioned the diagram (in another context), I love and use it - Talking about the diagram which is showed (on demand) before tour start and displaying the whole trip where x is the altitude and y the distance with the waypoints in between. Anyway, when navigation starts, I see only the section between two waypoints. Question: will it be possible to switch by selection or tapping during navigation between both diagrams?  - And on top as goodie to enter a rolling time i.e. 10sec. that the diagram switches every 10sec?

Speed Limit: I know it can be displayed. Will it be possible to display it as a real sign (being more visible) like you have it in other navigation apps (i.e. google maps) 

Best regards, Dirk

**Reply by devemux86** [reply]
@DIrgi 

- Select "Destination and waypoint" in "Settings | Navigation | Display | Distance / Time"
to show both of them at the same time. Also long press waypoint panel to select the waypoint.

- Odometer: we already measure and remember the average speed, we can measure more data.
Note: GPS recording can also be used to display the traveled distance and record the track.

- Diagram: when navigation starts, the diagram shows by default some km ahead of user location.
You can **zoom out** (swipe like map) the diagram so it displays more distance. Is it what you mean?

- Speed limit: now the current speed limit appears at the right panels and the warning as a sign.
We have a discussion for an alternative appearance, please see: https://github.com/devemux86/cruiser/discussions/452

**Comment by DIrgi** [comment]
* Select "Destination and waypoint" in "Settings | Navigation | Display | Distance / Time"

Sorry my fault, I asked wrong - what I meant and probably is not available is showing i.e. 
duration and final arriving time at the same time - I only can switch in-between when it comes to time.

* ODOmeter: we already measure and remember the average speed, we can measure more data.

Right but not like a continuously ODO meter afaik.

>   Note: GPS recording can also be used to display the traveled distance and record the track.

I  know but then I have to use the recording.

> * Diagram: when navigation starts, the diagram shows by default some km ahead of user location.
>   You can **zoom out** (swipe like map) the diagram so it displays more distance. Is it what you mean?

Exactly - cool...I tried it on my way but didn't work because I tried it "in the wrong way" I tried to zoom in/out by moving my fingers more like vertical or oblique but not horizontal - horizontal works - great.


> * Speed limit: now the current speed limit appears at the right panels and the warning as a sign.
>   We have a discussion for an alternative appearance, please see: [Speed limit display #452](https://github.com/devemux86/cruiser/discussions/452)

Thanks, I just overflew it - So the discussion is there alright - However - Can it be that sometimes the sign is not displayed and just the limit in the field in the right panel?

**Reply by devemux86** [reply]
>  showing i.e. duration and final arriving time at the same time

Indeed, we can add this option too.

> Can it be that sometimes the sign is not displayed and just the limit in the field in the right panel?

Do you mean that want only the current speed limit at the right panel,
but not the red warning sign when be above the speed limit?

This can be controlled in "Settings | Display | Speed limit", select:
- [x] Display
- [ ] Warning

**Comment by DIrgi** [comment]
> > showing i.e. duration and final arriving time at the same time
> 
> Indeed, we can add this option too. 

Cool.

> > Can it be that sometimes the sign is not displayed and just the limit in the field in the right panel?
> 
> Do you mean that want only the current speed limit at the right panel, but not the red warning sign when be above the speed limit?
> 
> This can be controlled in "Settings | Display | Speed limit", select:
> 
> * [x]  Display
> * [ ]  Warning

Yes I'm aware about this settings and I have all time display activated(only). I recognized that sometimes speed limit is generally not displayed neither as sign nor in the panel - I assume the information might then be missing in the map base material in general - or it's because of I'm cruising right now i.e. in a city where the limit is in general 50km/h except the exceptions.
But what I meant was that when the speed limit is displayed in the panel the additional red sign is not always shown at the same time and is sometimes missing - Am I right or can i be?

**Reply by WolfgangZ42** [reply]
> the additional red sign is not always shown

The "red sign" is just a warning, therefore its shown only if you are above the speed limit plus /Settings/Navigation/Advanced/Speed limit tolerance ... 
To display the current speed limit permanently I use Tempo Master (an overlaying app):
<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/be013e99-1528-4a63-b4cb-01b9abc9c05d"/>

**Reply by devemux86** [reply]
>  I assume the information might then be missing in the map base material

Correctly, the speed limit display at right panels appears when there is the data in OpenStreetMap database.

> the additional red sign is not always shown

This is the speed limit warning. It appears when current speed is above the speed limit (by the selected tolerance).

**Reply by 0709wiwiwi** [reply]
@WolfgangZ42  advice is most optimal considering TempoMaster (database) works even in the simple tracking mode.
Max speed storage in export files is thus unnecessary.  Cruiser routing and navigation optimisations are more important.

**Reply by devemux86** [reply]
@0709wiwiwi @WolfgangZ42 

I can implement a similar feature in Cruiser so that the speed limits are always available,
whether we are on or off route.

**Reply by DIrgi** [reply]
This would be nice since speed limit is changing quite often and then the government wants to "earn easy money" ;-)
It's good to have it more visible (on demand) in the screen. Thank you Emux for thinking about.

**Comment by Ciutnik** [comment]
@DIrgi touched on the difficulty of scaling a diagram with two fingers. I also experienced this, especially when wearing gloves it is troublesome. 

Maybe it would be enough to touch the diagram on the right to reduce its zoom (show a larger section of the diagram) and on the left to change the scale back? Sliding your finger would still move the diagram sideways.

**Reply by DIrgi** [reply]
Hi Ciutnik, 
yes I agree with you especially the handling by pulling/pushing the fingers in horizontal direction to zoom in and out is even a little bit more inconvenient instead of doing it vertical. The next thing is: On my tablet the diagram has a height of 2cm so it's still slim but I can touch it with a little bit of exercise (with gloves). I wonder how small it'll be on a smaller phone.
Finally I don't know if Emux will/or can change the gesture for zooming or attach a button for in/out zooming.
Depending on the programming effort. I can life with that on his decision, but for sure I'd love as well to have a slightly better access or even (what I mention) having a rolling diagram (on demand) switching i.e. every 10sec between close and total view.

As you read in my comment I was first of the opinion that I cannot change the zoom of the diagram until Emux told me how to do it.
So this was already a benefit and there I come again for my request for a manual it would help a lot, because on some things your are not even aware about that they are in.
But I understand his arguments - manuals are a lot of work and need to be done and adapted.
On the other hand it would prevent him from "stupid questions" like from my side - He could always say first - rtfm 😉

**Reply by devemux86** [reply]
We can try:
- Tap diagram on the **right** to zoom **in**
- Tap diagram on the **left** to zoom **out**

The zoom stays (like now) so you can adjust it anytime and keep traveling with the preferred diagram zoom.

**Reply by Ciutnik** [reply]
To make it not a "hidden feature", maybe it would be possible to place a slightly marked "+" and "-" big sign? Maybe just the outline? They may have been correlated with the show/hide on-screen buttons setting 🙂

**Reply by Ciutnik** [reply]
Sorry for the off-topic, regarding on-screen buttons: we once talked about making the compass the same size as the buttons. I guess it should be (?).

**Reply by DIrgi** [reply]
I think Emux had a good idea. 
Ciutnik Symbols would be great but...they will be some more symbols which cover something so sometimes is less, more. If Emux defines maybe that if you press somewhere in the 20% left half of the diagram it zooms in and in the 20% right half it zooms out (the "virtual" button behind is even bigger then) that would be fine enough - or? - And if you don't touch, nothing happens.

**Reply by devemux86** [reply]
We should be careful as the diagram has a small part of the screen (especially in navigation)
and it shows data and many labels. Adding more elements could make it not easily readable.

About the buttons vs compass size, most apps show smaller compass (that can be pressed).
Or the buttons should be the same compass size to not hide the screen. :slightly_smiling_face:

**Reply by devemux86** [reply]
Now it is 50% - 50% for zoom out / in.
We can adjust it to only 25% far left and 25% far right or other percentage if we like.

**Reply by Ciutnik** [reply]
Indeed, I agree with you 100%, personally I would not like to see additional symbols on the chart either. I meant only for newbies, but you only need to learn about this feature once. There is no need for constant reminders 👍

As for the size of the virtual keys, I would opt for about 25-33%. I would rather reserve the middle for unsuccessful attempts to move the chart sideways 🙂 Touches, especially swipes with a gloved finger, can be unsuccessful...

**Reply by devemux86** [reply]
We can do the same for the diagram in route planning.
However there the tap (and drag) is used to move the diagram cursor and the map center.

Or we could use the double tap to zoom in / out.

**Reply by devemux86** [reply]
What default distance ahead of user cursor should the navigation diagram shows?
Now it is 20 km for the driving profiles.

**Reply by DIrgi** [reply]
you mean the close up? - I would keep it, it's a good base.

**Reply by devemux86** [reply]
I meant with what default distance the navigation graph should start?
Now it starts on user location and zooms in to show 20 km ahead of it.

Is 20 km ok or we need something else?

**Reply by DIrgi** [reply]
Exactly, yes - Keep this it's a good base. Since you build a zooming mode, everybody can later on desire where to go. 
Or....offer the ability in the settings, where the user can self define on which zoom level he'll start - but again by default I would remain with 20km

**Reply by DIrgi** [reply]
@devemux86  It looks like that with cruiser-3.0.43-b9
you already had a hand on it. At least zooming in/out by tapping on the left/right area of the diagram will work. 👍
(red = tapping areas for zooming in/out)
![image](https://github.com/devemux86/cruiser/assets/57968526/6d98a362-240c-4eb2-80cb-a1182a2e5a1a)

1st
Question: Will it be possible that not only tapping but i.e. leaving the finger on the left/right side of the diagram starts an automatic zooming in/out after i.e. one second an the automatic step one level in/out each 20ms (has to be checked if smooth or to slow or fast.

2nd:
Question: Will it be possible to either/or (or even permanent for all visible waypoints on demand) by tapping on a waypoint in the diagram to see the name. I did an example construction how it can look like.
![image](https://github.com/devemux86/cruiser/assets/57968526/59456d62-116f-4a89-b3cd-17c2efaa12d5)

I know it'll be small, especially when you have a small smartphone. On a tablet it'll work. However in the example you see the name box examples popping up by tapping on a waypoint or are even visible the whole time (on settings demand as mentioned). A good orientation for the box can be, when the waypoint sign is below the middle line, the box appears above, when the symbol is above, then the box appears below.
What do you think?

**Reply by devemux86** [reply]
@DIrgi thanks for the suggestions.

1. Do you mean continuous zoom like when we press and hold the map zoom buttons?
I don't know if the diagram allows such gestures, I will check it.

2. It also depends on what the diagram allows to implement, as this is not a map library with rich overlays.
So touchable markers or custom labels may be too much for it.

About 2 please create a new topic, as here it could be forgotten. :slightly_smiling_face:

**Reply by DIrgi** [reply]
on 1 - That's fine if you can check it
on 2 - I moved it into a standalone request - See here 
https://github.com/devemux86/cruiser/discussions/505

**Comment by devemux86** [comment]
@DIrgi

Some small notes regarding the favorites in your excellent guides:
- To import as new group, select "Import" in any existing group menu,
import the gpx file and select "New group" in the import dialog
- We can also import zip files in favorites

**Reply by DIrgi** [reply]
- The zip thing is good to know. I'll keep it in mind.
- The thing with the group - Since it's a description and probably there's a more direct way I figured out that it can easily happen, when you are not trained "and press the wrong button" that you import the POIs without a group or at least into the wrong group. Then you have to delete everything again. So my estimation is, just keep this as a safe installation guide

btw. I don't know if I can write private messages here. Never the less. Do you think it can be possible to load and use my own overlay?
So the same I do with POIs in this case then with overlaid road? - Reason/idea behind. Marking the complete route which is closed in another color overlay or marking a rout which is very scenic and so on.

**Reply by devemux86** [reply]
> So my estimation is, just keep this as a safe installation guide

:+1: 

> So the same I do with POIs in this case then with overlaid road?

There are the favorite "Routes" and "Tracks":
- **Routes**: store complete routes with all navigation content, that can be loaded offline for navigation

- **Tracks**: store GPX tracks as collection of path coordinates (similar to recorded tracks)

Both can be visible on / off, with custom colors and zoom level.

**Reply by DIrgi** [reply]
Cool I just tested it...well you have to think slightly around the backdoor but this I can use to complete my POIs ;-)

**Reply by bios4** [reply]
> Marking the complete route which is closed in another color overlay or marking a rout which is very scenic and so on.

I created GPX tracks (using BRouter web) for the closed roads in Austria using the POIs from your collection as start/end points. These were then added to "My collection" in a separate "Closed roads" group, colored in the (same) red as the favorite POIs and are shown on my device as "overlay".

That's what it looks like now:

<img src="https://github.com/devemux86/cruiser/assets/145460147/83104bc8-8167-40e0-aa44-e907f98ef16b">

Scenic routes (I called them "WOW-routes", potentially including "WOW-spots") are added to "My collection" in a smilar way, colored differently:

<img src="https://github.com/devemux86/cruiser/assets/145460147/6633127d-e814-4359-8d5e-a8381978c026">

**Reply by DIrgi** [reply]
@bios4 

Mike exactly - I did meanwhile the same, Emux gave me the hint today. I will add this to my road closure POI publication. Maybe latest tomorrow.
btw. if you have good i.e. good POIs to share, maybe you like to share it here?
https://github.com/devemux86/cruiser/discussions/488
